tango/integrations/datasets/__init__.py

```diff
@@ -72,22 +101,65 @@ class LoadDataset(Step):
 
         Registered as a :class:`~tango.step.Step` under the name "datasets::load".
 
+    .. important::
+
+        If you are loading an :class:`~datasets.IterableDataset` or :class:`~datasets.IterableDatasetDict`
+        you need to use the :class:`LoadStreamingDataset` step instead.
+
     """
 
     DETERMINISTIC = True
     VERSION = "001"
-    CACHEABLE = False  # These are already cached by huggingface.
+    CACHEABLE = True
+    FORMAT = DatasetsFormat()
+
+    def run(self, path: str, **kwargs) -> Union[ds.DatasetDict, ds.Dataset]:  # type: ignore
+        """
+        Load the HuggingFace dataset specified by ``path``.
+
+        ``path`` is the canonical name or path to the dataset. Additional key word arguments
+        are passed as-is to :func:`datasets.load_dataset()`.
+        """
+        dataset = ds.load_dataset(path, **kwargs)
+        if not isinstance(dataset, (ds.Dataset, ds.DatasetDict)):
+            raise ConfigurationError(
+                f"{self.__class__.__name__} can only be used with non-streaming datasets. "
+                f"For streaming datasets, use the 'LoadStreamingDataset' ('datasets::load_streaming') step instead."
+            )
+        return dataset
+
+
+@Step.register("datasets::load_streaming")
+class LoadStreamingDataset(Step):
+    """
+    This step loads an iterable/streaming `HuggingFace dataset <https://huggingface.co/datasets>`_.
+
+    .. tip::
+
+        Registered as a :class:`~tango.step.Step` under the name "datasets::load_streaming".
+
+    """
+
+    DETERMINISTIC = True
+    VERSION = "001"
+    CACHEABLE = False  # can't be cached
```

Comment:
To be sure, Tango can serialize and deserialize iterators. There is support for that in `DillFormat` and I think also `JsonFormat`. But it's not clear that this is necessary in this case.

Reply:
I think that would be really inefficient, and a streaming dataset could potentially go on forever.

Reply:
32face0